src/loader.rs

```diff
@@ -59,17 +67,32 @@ impl LoaderInput {
                 }
             },
 
-            Self::PasswordLiteral(password) => Ok(password.to_string()),
-
-            Self::MnemonicPasswordLiteral(_, password) => Ok(password.to_string()),
+            Self::PasswordLiteral(password)
+            | Self::MnemonicPasswordLiteral(_, password)
+            | Self::RecoveryLiteral(_, password) => Ok(password.to_string()),
         }
     }
 
-    fn read_password<L: Ledger>(&mut self) -> Result<String, WalletError<L>> {
+    fn read_password<L: Ledger>(&mut self) -> Result<Option<String>, WalletError<L>> {
         match self {
-            Self::User(reader) => reader.read_password("Enter password: "),
-            Self::PasswordLiteral(password) => Ok(password.to_string()),
-            Self::MnemonicPasswordLiteral(_, password) => Ok(password.to_string()),
+            Self::User(reader) => loop {
+                println!("Do you remember your password? [y/n]");
```

Comment:
Oh I just noticed that the title of this PR was for "recovery/password reset", so perhaps change this message to "Do you forget your password or want to reset it?", then swap the logic for "yes" and "no".